builtin/providers/openstack/config.go

```diff
@@ -0,0 +1,67 @@
+package openstack
+
+import (
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack"
+)
+
+type Config struct {
+	Username         string
+	UserID           string
+	Password         string
+	APIKey           string
+	IdentityEndpoint string
+	TenantID         string
+	TenantName       string
+	DomainID         string
+	DomainName       string
+
+	osClient *gophercloud.ProviderClient
+}
+
+func (c *Config) loadAndValidate() error {
+	ao := gophercloud.AuthOptions{
+		Username:         c.Username,
+		UserID:           c.UserID,
+		Password:         c.Password,
+		APIKey:           c.APIKey,
+		IdentityEndpoint: c.IdentityEndpoint,
+		TenantID:         c.TenantID,
+		TenantName:       c.TenantName,
+		DomainID:         c.DomainID,
+		DomainName:       c.DomainName,
+	}
+
+	client, err := openstack.AuthenticatedClient(ao)
+	if err != nil {
+		return err
+	}
+
+	c.osClient = client
+
+	return nil
+}
+
+func (c *Config) blockStorageV1Client(region string) (*gophercloud.ServiceClient, error) {
+	return openstack.NewBlockStorageV1(c.osClient, gophercloud.EndpointOpts{
+		Region: region,
+	})
+}
```

Comment:
@phinze It sounds like you're suggesting that there be one client per service in the `Config` struct, however many OpenStack services are region-specific, so unless each resource gets its own copy of the `Config` struct, I'm not sure that'd work. For example, given a compute resource that's to be built in Region1 and a second compute resource that's to be built in Region2, then 2 different compute clients are needed to create those resources.